app/HutchManager/Models/ManageAgent.cs

```diff
@@ -0,0 +1,26 @@
+using HutchManager.Extensions;
+
+namespace HutchManager.Models;
+
+public class ManageAgent
+{
+  public int Id { get; set; }
+
+  public string Name { get; set; } = string.Empty;
+
+  public string ClientId { get; set; } = string.Empty;
+
+  public string ClientSecretHash { get; set; } = string.Empty;
+
+  public List<Data.Entities.DataSource> DataSources { get; set; } = new();
+  
+   public Data.Entities.Agent ToEntity()
```

Comment:
can we remove `ToEntity()` in favour of mapping models to entities in the Service Method, where it's clearer?